### PR TITLE
Added support for kerberos authentication

### PR DESCRIPTION
**Pull request by jatinderjawanda**
Install krb5-user pkg to have kerberos support

Signed-off-by: JatinderJawanda <jatinder.singh@guavus.com>

### CODE REVIEW

**Comment by jatinderjawanda** [comment]
> Committer of one or more commits is not listed as a CLA signer

done

**Comment by FlorianHockmann** [comment]
Hi @jatinderjawanda and thanks for your contribution!
I'm wondering though whether we should include support for Kerberos in JanusGraph's images. Any reason why you can't just add that in your own image that depends on the official one from JanusGraph?

**Comment by jatinderjawanda** [comment]
> Hi @jatinderjawanda and thanks for your contribution!
> I'm wondering though whether we should include support for Kerberos in JanusGraph's images. Any reason why you can't just add that in your own image that depends on the official one from JanusGraph?

Hey @FlorianHockmann,
We already build our image with kerberos changes. I raised here as it will be add-on to existing functionality provided by image without increasing the image size at large. Furthermore, if someone want to use official image as it then they don't have to rebuild the image if kerberos support exists. So, I fathom, its good to have this support inbuilt into official image.

File: 0.4/Dockerfile [hunk 1]
```diff
@@ -32,10 +32,13 @@ ENV JANUS_VERSION=${JANUS_VERSION} \
     janusgraph.storage.directory=/var/lib/janusgraph/data \
     gremlinserver.graph=/etc/opt/janusgraph/janusgraph.properties \
     gremlinserver.threadPoolWorker=1 \
-    gremlinserver.gremlinPool=8
+    gremlinserver.gremlinPool=8 \
+    DEBIAN_FRONTEND=noninteractive
 
 RUN groupadd -r janusgraph --gid=999 && \
     useradd -r -g janusgraph --uid=999 janusgraph && \
+    apt-get update -yqq && \
```
Comment (FlorianHockmann) [inline 0.4/Dockerfile]: (nitpick) Do we really need to make this completely quiet? Wouldn't a bit of output be helpful, especially if this fails for some reason?

File: 0.4/Dockerfile [hunk 2]
```diff
@@ -32,10 +32,13 @@ ENV JANUS_VERSION=${JANUS_VERSION} \
...
+    apt-get install -yqq krb5-user && \
```
Comment (FlorianHockmann) [inline 0.4/Dockerfile]: Please add the following line below to clean up the apt cache afterwards to reduce image size:
```
rm -rf /var/lib/apt/lists/* && \
```

File: 0.4/Dockerfile [hunk 3]
```diff
@@ -32,10 +32,13 @@ ENV JANUS_VERSION=${JANUS_VERSION} \
     janusgraph.storage.directory=/var/lib/janusgraph/data \
     gremlinserver.graph=/etc/opt/janusgraph/janusgraph.properties \
     gremlinserver.threadPoolWorker=1 \
-    gremlinserver.gremlinPool=8
+    gremlinserver.gremlinPool=8 \
+    DEBIAN_FRONTEND=noninteractive
```
Comment (FlorianHockmann) [inline 0.4/Dockerfile]: Why was this change necessary? [The Docker docs recommend](https://docs.docker.com/engine/faq/#why-is-debian_frontendnoninteractive-discouraged-in-dockerfiles) against using this.

File: 0.4/Dockerfile (hunk 1 above)
Comment (jatinderjawanda) [reply 0.4/Dockerfile]: Done. Removed quiet flags.

> You need [Kerberos for Solr](https://docs.janusgraph.org/index-backend/solr/#kerberos-configuration), right? A comment directly in the `Dockerfile` could also be a good idea here to describe why this dependency is needed for future contributors.
> If a backend we ship needs such a dependency for a certain functionality, then I think it's OK to add it.
> 
> I added a few minor comments. Apart from that, please don't edit the `Dockerfile` by hand as it's generated automatically from the `bulid/Dockerfile-openjdk8.template` as indicated in its header. Instead, modify the template and then run the `update.sh` script to generate the `Dockerfiles` for the different versions.

We need kerberos support for any secured backend storage.(In our case, HBase).

Thanks for heads up !! made changes via docker template.

File: 0.4/Dockerfile (hunk 2 above)
Comment (jatinderjawanda) [reply 0.4/Dockerfile]: Done.

File: 0.4/Dockerfile (hunk 3 above)
Comment (jatinderjawanda) [reply 0.4/Dockerfile]: Removed from ENV and added it to specific command as if this `DEBIAN_FRONTEND=noninteractive` not provided then installation of krb5-user prompts kerberos related configs like realm.

File: 0.4/Dockerfile (hunk 1 above)
Comment (FlorianHockmann) [reply 0.4/Dockerfile]: Thanks! The changes in the template now look good to me in general, but could you please execute `update.sh`? This will update this `Dockerfile` and the one in `0.3`.

File: 0.4/Dockerfile (hunk 3 above)
Comment (FlorianHockmann) [reply 0.4/Dockerfile]: OK, good to know and the solution you used now here sounds good to me.

**Comment by FlorianHockmann** [comment]
Apart from running `update.sh` to generate the `Dockerfiles` again, could you please also squash your git commits? We prefer to have contributions in a single commit.

File: 0.4/Dockerfile (hunk 1 above)
Comment (jatinderjawanda) [reply 0.4/Dockerfile]: done

**Comment by jatinderjawanda** [comment]
@FlorianHockmann : Raised another PR with all review comments incorporated.
